v3/memo.go

```diff
@@ -207,6 +219,26 @@ type memoGroup struct {
 	// The scalar properties for the group. Nil if the group contains relational
 	// expressions.
 	scalarProps *scalarProps
+	// Map from optimization context (i.e. required physicalProperties)
+	// fingerprint to optimization state (the best plan and cost, the children
+	// locations associated with that plan, etc).
+	//
+	// TODO(peter): We intern the physicalProps in memo, which should allow this
+	// to be a map[*physicalProps]exprID.
+	optMap map[string]*memoOptState
```

Comment:
How will you handle "wildcard" matches? For example, a parent expression might require a particular child sort order, but at the same time not care about the value of a different physical property, like distribution.

Reply:
As far as I can tell from my readings and looking at Orca, we need to worry about exact match. I should probably add a TODO here to understand this better as my intuition is that we'd need wildcard matches as well.